src/fpm_source_parsing.f90

```diff
@@ -77,10 +77,18 @@ function parse_f_source(f_filename,error) result(f_source)
     type(error_t), allocatable, intent(out) :: error
 
     integer :: stat
+    logical :: exists
     integer :: fh, n_use, n_include, n_mod, i, j, ic, pass
     type(string_t), allocatable :: file_lines(:), file_lines_lower(:)
     character(:), allocatable :: temp_string, mod_name, string_parts(:)
 
+    ! check the f_filename exists before attempting to process 
+	inquire(file=f_filename, exist=exists)
+	if (.not. exists) then
+		call file_not_found_error(error, f_filename)
+		return
+	end if
```

Comment:
```suggestion
    inquire(file=f_filename, exist=exists)
    if (.not. exists) then
        call file_not_found_error(error, f_filename)
        return
    end if
```

Reply:
Thanks for fixing the code formatting. Is there a recommended Fortran code formatting tool that is generally used by the fortran-lang projects to ensure consistency?

Reply:
fprettify usually does a good job.